locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/JcloudsProviderAndApiLoader.java

```diff
@@ -84,13 +85,15 @@ public static boolean isApi(String id) {
     }
     
     public static Optional<ProviderMetadata> getProvider(String id) {
+        id = DeserializingJcloudsRenamesProvider.getInstance().applyJcloudsRenames(id);
```

Comment:
I wonder if we should overwrite the provider config key with the new name. Otherwise, we'll need to leave these values in the properties file forever. But I'm fine with us not doing that in this PR.